### PR TITLE
[e2e] Check email for cancelled order

### PR DESCRIPTION
**Pull request by Stojdza**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes https://github.com/woocommerce/woocommerce-quality/issues/655

Using the similar approach as in [checking new order emails](https://github.com/woocommerce/woocommerce/blob/107bf38f585253e891393b4d97ae152a19fb5ee7/plugins/woocommerce/tests/e2e-pw/tests/merchant/order-emails.spec.js#L66), I added a test which cancels the order and checks if the email is received. 

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Run `pnpm test:e2e-pw ./tests/e2e-pw/tests/merchant/order-emails.spec.js -g "can receive cancelled order email" --headed` and verify is passed. 
2. Code review and verify approach. 
3. Make sure that CI jobs are ✅ 

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

Add new "can receive cancelled order email" test

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/tests/e2e-pw/tests/merchant/order-emails.spec.js [hunk 1]
```diff
@@ -94,6 +97,37 @@ test.describe( 'Merchant > Order Action emails received', () => {
...
+		await page.goto(
+			`wp-admin/tools.php?page=wpml_plugin_log&s=${ encodeURIComponent(
+				customerBilling.email
+			) }`
+		);
+		await expect(
+			page.getByText(
+				`[${ storeName }]: Order #${ cancelledOrderId } has been cancelled`
+			)
+		).toBeVisible();
```
Comment (Stojdza) [inline plugins/woocommerce/tests/e2e-pw/tests/merchant/order-emails.spec.js:129]: Hi @lanej0 👋 This same approach can be used in the test above, `can receive new order email`. If this looks good to you and the team, I can also update other tests.

**Comment by Stojdza** [comment]
@woocommerce/solaris @woocommerce/vortex 👋 I would appreciate some 👀 on this one so it doesn't get stale. Thanks!

**Review by veljkho — APPROVED** [review]
Looks good to me. However, I am not a fan of using API to trigger something like this; I would rather go to the existing order and edit it to change its status to from processing to canceled and then wait for the email.

At this stage, this seems enough, as other tests are also created the same way. 👍

**Review by rrennick — APPROVED** [review]
LGTM

**Comment by Stojdza** [comment]
> Looks good to me. However, I am not a fan of using API to trigger something like this; I would rather go to the existing order and edit it to change its status to from processing to canceled and then wait for the email.

Thank you @veljkho for bringing this up. This is a broader topic that we can discuss on a p2 with the team and we can make some decisions on how to approach this in the future

**Pull request merged**